### DISCUSSION TITLE
coffee instal path changes (?)

### DISCUSSION THREAD

**Discussion by vincenzopalazzo** [coffee-tools/coffee#259] [Q&A] [answered]
Currently, we install a coffee plugin inside a custom path `<networ>/plugins` but now that we merge https://github.com/coffee-tools/coffee/pull/239 still make sense?

I think we can remove this extra step of complexity and move to use the repositories directory? or does the plugin directory still make sense in the case of a big repository with multiple plugins?

**Comment by tareknaser** [comment]
I think it's overall better design that we have plugins installed in a separate directory

> does the plugin directory still make sense in the case of a big repository with multiple plugins?

Exactly. 
For example, this way we can have commands such as `coffee upgrade <plugin>` (We upgrade only one plugin from the remote repository without affecting other plugins in the same repository)

However, I am not sure we should even support the upgrade command on plugins level. It's more intuitive that the upgrade command works on repositories not plugins. We have a tracking issue https://github.com/coffee-tools/coffee/issues/187

**Reply by vincenzopalazzo** [reply]
>However, I am not sure we should even support the upgrade command on plugin level. It's more intuitive that the upgrade command works on repositories, not plugins. We have a tracking issue https://github.com/coffee-tools/coffee/issues/187

Well, my opinion is that the upgrade command should be by default on the plugin level, and then we have the possibility to upgrade everything (repo included) with some sort of command.

Usually, you want to manage the plugin and not the repository. The repository is only our concept of managing plugins 

What do you think? why in your opinion the upgrade should work on a repository base case? 

What is your workflow to upgrade a plugin?

**Reply by tareknaser** [reply, marked as answer]
> Usually, you want to manage the plugin and not the repository. The repository is only our concept of managing plugins

That does indeed make more sense.
I agree. Let's have `coffee upgrade` default to plugins. It adds to a better user experience.
